### PR TITLE
Feat: automate external doc updates

### PR DESCRIPTION
**Pull request by juliamrch**
## Describe your PR

A Clever Cloud guide has been included on Astro, and therefore we updated as well our guide to include detailed configuration depending on whether it's deployed on a static or a node application.

In order to avoid asynchronicity between our development and the external doc posted on diverse frameworks, I wrote an automation to check if the modified files are linked to some external doc. If relevant, the bot creates an issue and assign it to the PR author, as well as notifying this on a comment.  

- [Updated Astro guide](http://documentation-pr-388.cleverapps.io/guides/astro/)
-  [Demo: Issue created](https://github.com/juliamrch/cc-doc-tests/issues/30)
- [Demo: Comment on PR](https://github.com/juliamrch/cc-doc-tests/pull/29#issuecomment-2387222892)

## Checklist

- [x] My PR is related to an opened issue : #383
- [x] I've read the [contributing guidelines](/CleverCloud/documentation/blob/main/CONTRIBUTING.md)

## Reviewers
_Who should review these changes?_ @cnivolle

### CODE REVIEW

File: content/guides/astro.md [hunk 1]
```diff
@@ -17,11 +17,172 @@ comments: false
...
+      }
+    });
+    ```
+
+  {{< /tab >}}
+
+
+{{< /tabs >}}
+
+## Deploy Astro from the Console
```
Comment (davlgd) [inline content/guides/astro.md:70]: This kind of instructions are more a "How to use Console" guide than a "How to deploy this tech/framework on Clever Cloud" guide. It's why I usually detail only CLI instructions which show what's needed/important (and because the user can just copy/paste the code and see it deploying). We could refer to Console documentation for those needing instructions about it.

File: content/guides/astro.md [hunk 2]
```diff
@@ -39,4 +200,13 @@ clever env set CC_PRE_BUILD_HOOK "npm install && npm run astro telemetry disable
...
-{{% content/git-push %}}
\ No newline at end of file
+{{% content/git-push %}}
+
+## Learn more
+
+{{< cards >}}
+  {{< card link="../doc/applications/javascript/nodejs" title="Deploy a Node.js application" subtitle="Learn more on deploying a Node.js application on Clever Cloud" icon="node" >}}
+  {{< card link="../doc/applications/static" title="Deploy a Static application" subtitle="Learn more on deploying a Static application on Clever Cloud" icon="feather" >}}
+  {{< card link="https://docs.astro.build/en/getting-started/" title="Learn Astro" subtitle="Astro full documentation" icon="rocket-launch" >}}
```
Comment (davlgd) [inline content/guides/astro.md:210]:
```suggestion
  {{< card link="https://docs.astro.build/" title="Learn Astro" subtitle="Astro full documentation" icon="rocket-launch" >}}
```

The more generic an external link is, the less we'll need to update it ^^

File: content/guides/astro.md [hunk 3]
```diff
@@ -17,11 +17,172 @@ comments: false
...
+
+   ```json {filename="package.json"}
+    "scripts": {
+      "dev": "astro dev",
+      "start": "astro dev",
+      "build": "astro check && astro build",
+      "preview": "astro preview --host 0.0.0.0 --port 8080",
+      "astro": "astro"
+    } 
+    ```
```
Comment (davlgd) [inline content/guides/astro.md:49]: Some parts of the file are not needed, start for example as we use a `CC_RUN_COMMAND`

File: content/guides/astro.md [hunk 4]
```diff
@@ -17,11 +17,172 @@ comments: false
...
+
+{{< tabs items="pnpm,yarn" >}}
+  
+  {{< tab >}}
+
+  ```shell
+    CC_NODE_BUILD_TOOL="custom"
+    CC_PRE_BUILD_HOOK="npm install -g pnpm && pnpm install"
+    CC_CUSTOM_BUILD_TOOL="pnpm run astro telemetry disable && pnpm build"
+    CC_RUN_COMMAND="pnpm run preview"
```
Comment (davlgd) [inline content/guides/astro.md:120]: From [Astro documentation](https://docs.astro.build/en/reference/cli-reference/#astro-preview) regarding the `preview` command: 

> This command allows you to preview your site locally [after building](https://docs.astro.build/en/reference/cli-reference/#astro-build) to catch any errors in your build output before deploying it. It is not designed to be run in production.

**Review by davlgd — COMMENTED** [review]
Thanks for this. 

I didn't review the action part, only the guide.

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,5 @@
 .DS_Store
+.idea
```
[the remark this answers is not in the document]
Comment (juliamrch) [reply .gitignore:2]: Yes, it's just the config files for IntelliJ, some people use it and it ignores if they write the doc from it

File: content/guides/astro.md (hunk 1 above)
Comment (juliamrch) [reply content/guides/astro.md:70]: It's a copy from the Astro guide, which of course will be more explicit to provide a glance of the Console. However, we need to keep in mind that people reading our guides will probably do so from some Google search. Imo, guides need to be as complete as possible, even if it means including beginner-like content, because it can be the only thing people will read from our doc, just to deploy a framework. The more step-by-step content, the better, since from a single page, they'll be able to deploy their site.

If we add too many reference on out guides page, the risk of people thinking "man, now I got to learn how the damn console works, oh and with is the CLI" increases. Guides are meant to be a one-shot page instructions from start to end, found on a search, complete enough to improve SEO with quality content and reduce the learning curve of Clever Cloud deployment. A single guide can replace many pages of this doc, and make someone competent enough to deploy without needing further instructions.

TL; DR: baby steps are staying

File: content/guides/astro.md (hunk 2 above)
Comment (juliamrch) [reply content/guides/astro.md:210]: true!

File: content/guides/astro.md (hunk 3 above)
Comment (juliamrch) [reply content/guides/astro.md:49]: It can be set up both ways.

File: content/guides/astro.md (hunk 4 above)
Comment (juliamrch) [reply content/guides/astro.md:120]: Exactly [what I submitted](https://github.com/withastro/docs/pull/9030#discussion_r1752270277) to @TheOtterlord but it seems that no command seems to be designed to run on production? I'm confused as well lol

Comment (davlgd) [reply content/guides/astro.md:120]: Astro is not made to serve the website directly. It's why it's often used to build them and then it's served with a dedicated static web server (Apache on Clever Cloud for now in the static runtime for example), or through adapters in SSR.

Comment (TheOtterlord) [reply content/guides/astro.md:120]: the server from the node adapter can be run directly with `node ./dist/server/entry.mjs`

File: content/guides/astro.md (hunk 1 above)
Comment (juliamrch) [reply content/guides/astro.md:70]: >  (developers) can handle to use a node CLI

Let's not assume what our targeted audience can manage /will be willing to use or not. Both Console-focused and CLI-focused instructions should be able to fit into guides, and users will choose whichever option suits them better.

But I see your point. I propose that we follow the tunnel creation of the Console for the steps, without detailing it so much (it's a little bit over the top since it was made for Astro doc), and that we bring the following changes for steps : 

1. Create the application: summarizes the deployment, the type, and the size, with relevant links if needed
2. Connected Services
3.  Inject environment variables

Comment (davlgd) [reply content/guides/astro.md:70]: > Both Console-focused and CLI-focused instructions should be able to fit into guides, and users will choose whichever option suits them better.

We agree on this. So I think the better is to keep Console instructions, but simplify as you mentioned in Option 1 (and don't list a 7 steps process), focus on what's specific to Astro static deployment. And to Add a CLI section using existing instructions (including build cache stuff in both, maybe as an optional step). 

One point of attention on the Console instructions: as the whole creation process is being revamped, we'll have to update them after release. The more we detail it in each guide independently, parts of the doc, the more we'll have to make updates later.

For the SSR part discussed below, maybe we should just mention the user can use the Node adapter and send [to Astro doc](https://docs.astro.build/en/guides/integrations-guide/node/) about that in a dedicated section at the end, with instructions on what the user needs to do to activate it and the benefits.

I think one important message the reader should get when first seeing the page is: there is (almost) nothing special to do to deploy Astro on Clever Cloud. But if you want to go further (pnpm/yarn, build cache optimization, SSR), here's what you need to know.

File: content/guides/astro.md [hunk 6]
```diff
@@ -17,16 +17,164 @@ comments: false
...
+
+### Serve the site from server
+
+If the app uses the [Node.js adapter](https://docs.astro.build/en/guides/integrations-guide/node/), it can start by either:
+
+- Injecting `CC_RUN_COMMAND=./dist/server/entry.mjs` in environment variables. `CC_RUN_COMMAND` variable always overrides start script in `package.json`.
+- Setting `./dist/server/entry.mjs` as start script in `package.json`. 
+
+## Deploy an Astro site using the CLI
+
```
Comment (davlgd) [inline content/guides/astro.md:172]: I don't think it the good place for this. This remark is linked to other and the "Port and Host" part at the beginning of the guide. 

By default, and in the project used here, Astro build a static website. It is served from a server and a Clever Cloud user can host it using a static runtime, or another runtime with a static web server (in Node, Rust, Python, whatever). We recommend the static runtime as it's made for this (and will soon evolve to better respond to such needs). In this case, there is no specific Astro config to make, no port to define. And we can take profit of the build cache to just keep built website in it.

In the Astro config, the user can ask for the server to render all or some pages on-demand (SSR mode). `static` output should be replaced by `server` (dynamic website with some pre-rendered pages if asked) or `hybrid` (static website with some dynamic pages if asked).

In this case, if the Node adapter is installed and used, the server can be served with `node ./dist/server/entry.mjs` or using `./dist/server/entry.mjs` as a start script defined in `package.json`. But there are no case where `astro dev` or `astro preview` should be used to serve a website. It's only for dev/preview purposes.

All of this can be done on Clever Cloud through CLI or Console. 

So I think this part (and the host/port) should be in a SSR dedicated section after the static (simpler) one, explaining what are the differences/benefits of SSR (server or hybrid), the Node adapter, and what should be changed on the Astro config and env vars to use it. 

For pnpm/yarn, I think this make things more complex to understand when a user just want to deploy the provided example, and maybe should be detailed in a dedicated section. I let you decide.

**Review by davlgd — DISMISSED** [review]

File: .github/workflows/external-doc-update.yml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+on:
+  pull_request_target:
+    types: [opened, synchronize]
+    branches: [ main ]
+
+jobs:
+  changed_files:
+    runs-on: ubuntu-latest
+    name: Test changed-files
+    permissions: # Needed to post comments and create issues
```
Comment (HeleneAmouzou) [inline .github/workflows/external-doc-update.yml:12]:
```suggestion
    # Needed to post comments and create issues
    permissions:
```

File: .github/workflows/external-doc-update.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        uses: taoliujun/action-unique-comment@v1
+        with:
+          uniqueIdentifier: ${{ github.workflow }}
+          body: |
+            You updated ${{ env.CHANGED_FILES }}. This content is also listed on external doc. Issue number ${{ steps.create-issue.outputs.number }} has been created and assigned to you 🫵👁️👄👁️
+            
+            See it or modify it here: 
+              * ${{ steps.create-issue.outputs.url }}
+            
+            _This unique comment uses the very cool [taoliujun/action-unique-comment](https://github.com/marketplace/actions/unique-comment). Thank you <3_
```
Comment (HeleneAmouzou) [inline .github/workflows/external-doc-update.yml:62]: **praise:** I didn't know that tool, very usefull !

File: content/guides/astro.md [hunk 9]
```diff
@@ -17,16 +17,164 @@ comments: false
...
+    ```
+
+  {{< /tab >}}
+
+
+{{< /tabs >}}
+
+## Deploy Astro  from the Console
+
+To deploy your Astro project to Clever Cloud, you **create a new application**.
```
Comment (HeleneAmouzou) [inline content/guides/astro.md:73]:
```suggestion
To deploy your Astro project to Clever Cloud, you need to **create a new application**. 
```

**Review by HeleneAmouzou — DISMISSED** [review]
Nice work ! :clap: 
I've only left few suggestions, but other LGTM ! Also I've caught up your discussions with @davlgd and I agree with the decisions made.

File: content/guides/astro.md (hunk 6 above)
Comment (HeleneAmouzou) [reply content/guides/astro.md:172]: My bad, I didn't take into account that a default Astro project would be a static website and therefore can be deploy on Clever Cloud using static runtime or runtime with a static web server . Taking this in consideration and after going through other documentation, I agree that there should be a specific section for server rendering. So regarding this doc, maybe : first the most basic instructions for static Astro project, then specific instructions and explanations for SSR with the Port and host section, etc.

Otherwise I agree about the non usage of `astro dev` and `astro preview` in a deployment environment. 

About pnpm/yarn, I didn't feel like it was adding complexity, it is ok for me if it is not detailed in a specific section.

File: .github/workflows/external-doc-update.yml (hunk 8 above)
Comment (juliamrch) [reply .github/workflows/external-doc-update.yml:63]: It's very good, @taoliujun deserves credit on every comment <3

File: content/guides/astro.md (hunk 6 above)
Comment (juliamrch) [reply content/guides/astro.md:172]: Oops I resolved the wrong discussion, yes @HeleneAmouzou this was the input I needed, thank you for taking the time to review it!

File: content/guides/astro.md (hunk 1 above)
Comment (juliamrch) [reply content/guides/astro.md]: Settled for option 1 and SSR part being discussed above.

File: content/guides/astro.md [hunk 10]
```diff
@@ -17,16 +17,164 @@ comments: false
 draft: false
 ---
 
+{{< hextra/hero-subtitle >}}
+  Astro is an all-in-one web framework for building content-driven websites, that pioneers a new frontend architecture to reduce JavaScript overhead and complexity. Learn in this guide how to deploy an Astro site on Clever Cloud.
+{{< /hextra/hero-subtitle >}}
+
+## Overview
+
+Clever Cloud supports deploying both fully static and on-demand rendered Astro projects. Regardless of your `output` mode ([pre-rendered or on-demand](/en/basics/rendering-modes/)), you can choose to deploy as a **static application** which runs using a post-build hook, or as a **Node.js** application, which works out-of-the-box with your `package.json`.
```
Comment (juliamrch) [inline content/guides/astro.md:26]: As suggested by @sardemff7 in t[his commit](https://github.com/CleverCloud/documentation/commit/acd57a05b8a28934d4a94add7bbd2e6255e2412b#diff-f45ef70a1d05b0377e552d9bc23317bfdf8cfcf2f12bea7e41bfd46214ce27a5R37), because it summarizes nicely the purpose of each configuration.

Really love the package automation idea in your branch but I see some difficulty to implement it on guides, because often the project configuration is too specific to enable templating.

However, I think it's an idea we should explore.

```suggestion
Clever Cloud supports deploying both [fully static and on-demand rendered](https://docs.astro.build/en/basics/rendering-modes/) Astro projects:
- The `static` output mode is ideal for most content-oriented website, for which you have no need for per-visitor server-side customisation. Consider using a [Static runtime](/doc/applications/static/) when using this output mode, with the site generation in a post-build hook.
-  The `server` or `hybrid` output modes: consider using a [Node.js runtime](/doc/applications/javascript/nodejs/) runtime with [Astro’s Node adapter](https://docs.astro.build/en/guides/integrations-guide/node/)
```

File: content/guides/astro.md (hunk 6 above)
Comment (juliamrch) [reply content/guides/astro.md]: You both raised valid points regarding the deployment and clarity of the guide. These are the following points that I addressed in commit `4f5edb84`:
- Create a SSR section that includes host and port configuration
- Differentiate prod vs dev environment

The point about pnpm/yarn could add complexity, but as @HeleneAmouzou stated, it probably won't be complex for developers using these package managers. Astro users, and JS frameworks users in general, are widely familiar with them, and seeing a modern package manager as an option aligns the doc with the dynamism and the options available in this ecosystem.

File: content/guides/astro.md [hunk 11]
```diff
@@ -17,16 +17,130 @@ comments: false
...
+
+{{% steps %}}
+
+### Create the application
+
+In this step, you set up the following parameters:
+
+- Deployment (using Git or GitHub)
+- Application : Node.js or Static (both can serve a static site)
+- Instance size and scalability options : Astro sites can typically be deployed using the **Nano** instance
```
Comment (juliamrch) [inline content/guides/astro.md:45]:
```suggestion
- Instance size and scalability options : You can usually deploy an Astro sites using the **Nano** instance
```

File: content/guides/astro.md [hunk 12]
```diff
@@ -17,16 +17,130 @@ comments: false
...
+- Region
+- Dependencies, if needed
+
+### Inject environment variables
+
+Environment variables are used to control the deployment behavior, and depend on the type of application and your package manager. Find right below the ones that apply for your application.
```
Comment (juliamrch) [inline content/guides/astro.md:51]:
```suggestion
Use environment variables to control the deployment behavior, which depends on the type of application and your package manager. Find right below the ones that apply for your application.
```

File: content/guides/astro.md [hunk 13]
```diff
@@ -39,4 +153,72 @@ clever env set CC_PRE_BUILD_HOOK "npm install && npm run astro telemetry disable
...
+
+For SSR deployments, ensure to configure your application to listen on port **8080** as required by Clever Cloud. This differs from static deployments where such configurations are typically unnecessary.
+
+Set your port and host in your `astro dev` script for development mode, and/or configure it directly for production:
+
+{{< tabs items="dev,production" >}}
+
+  {{< tab >}}
+
+  To quickly deploy on dev mode:
```
Comment (juliamrch) [inline content/guides/astro.md:172]:
```suggestion
  To quickly deploy on development mode:
```

File: content/guides/astro.md [hunk 14]
```diff
@@ -39,4 +153,72 @@ clever env set CC_PRE_BUILD_HOOK "npm install && npm run astro telemetry disable
...
+
+To deploy an Astro SSR project, consider using a **Node.js** application on Clever Cloud. This is especially useful if your project uses the [Node.js adapter](https://docs.astro.build/en/guides/integrations-guide/node/).
+
+### Port and host
+
+For SSR deployments, ensure to configure your application to listen on port **8080** as required by Clever Cloud. This differs from static deployments where such configurations are typically unnecessary.
+
+Set your port and host in your `astro dev` script for development mode, and/or configure it directly for production:
+
+{{< tabs items="dev,production" >}}
```
Comment (juliamrch) [inline content/guides/astro.md:168]:
```suggestion
{{< tabs items="development,production" >}}
```

**Review by cnivolle — APPROVED** [review]
